src/ir_interpreter/interpret_ir.ml

```diff
@@ -498,16 +498,14 @@ and interpret_exp_mut env exp (k : V.value V.cont) =
     interpret_exp env' exp1 k
   | TryE (exp1, cases, Some (id, ty)) ->
     assert env.flavor.has_await;
-    let exp2 = Construct.(varE (var id ty)) in
+    let exp2 = Construct.(varE (var id ty) -*- unitE ()) in
     let k' v1 =
-      let cleanup v2 = interpret_exp env exp2 (fun _ -> k v2) in
-      interpret_catches env cases exp.at v1 cleanup in
-    let out ret v = interpret_exp env exp2 (fun _ -> ret v) in
-    let k'' v2 = interpret_exp env exp2 (fun _ -> k v2) in
+      interpret_catches env cases exp.at v1 k in
```

Comment:
Isn't it incorrect to skip the cleanup in the cases?

Reply:
Nope, as this is post-de sugaring which already inserted cleanups into the regular flow (i.e. after the entire `try .. catch`).

Reply:
This is moot now, code is rewritten.